packages/docregistry/src/context.ts

```diff
@@ -406,6 +409,11 @@ export class Context<T extends DocumentRegistry.IModel>
         };
       }
       this._path = newPath;
+      if (this.sessionContext.session) {
+        void this.sessionContext.session.setPath(newPath);
+      } else {
+        this.sessionContext.path = newPath;
+      }
```

Comment:
Should this replace the line below?

Reply:
Yes, thank you.

Reply:
Fixed, and I rebased to pull in the latest changes in CI testing.